workflow/artifacts/http/http.go

```diff
@@ -72,9 +72,41 @@ func (h *ArtifactDriver) Load(inputArtifact *wfv1.Artifact, path string) error {
 	return err
 }
 
-func (h *ArtifactDriver) OpenStream(a *wfv1.Artifact) (io.ReadCloser, error) {
-	// todo: this is a temporary implementation which loads file to disk first
-	return common.LoadToStream(a, h)
+func (h *ArtifactDriver) OpenStream(inputArtifact *wfv1.Artifact) (io.ReadCloser, error) {
+	var req *http.Request
+	var url string
+	var err error
+	if inputArtifact.Artifactory != nil && inputArtifact.HTTP == nil {
+		url = inputArtifact.Artifactory.URL
+		req, err = http.NewRequest(http.MethodGet, url, nil)
+		if err != nil {
+			return nil, err
+		}
+		req.SetBasicAuth(h.Username, h.Password)
+	} else {
+		url = inputArtifact.HTTP.URL
+		req, err = http.NewRequest(http.MethodGet, url, nil)
+		if err != nil {
+			return nil, err
+		}
+		for _, h := range inputArtifact.HTTP.Headers {
+			req.Header.Add(h.Name, h.Value)
+		}
+		if h.Username != "" && h.Password != "" {
+			req.SetBasicAuth(h.Username, h.Password)
+		}
```

Comment:
can we consolidate the logic that's copy/pasted from `Load()` into a single place?

Reply:
is the only actual difference the very last part in fact?

Reply:
Done. Yes there's only slight difference.

Reply:
Another difference is that the response body cannot be closed in `OpenStream()` since we will consume the stream later.